modules/core/include/opencv2/core/base.hpp

```diff
@@ -440,17 +440,17 @@ configurations while CV_DbgAssert is only retained in the Debug configuration.
 #endif
 
 #define CV_Assert_1 CV_Assert
-#define CV_Assert_2( expr1, expr2 ) CV_Assert_1(expr1); CV_Assert_1(expr2)
-#define CV_Assert_3( expr1, expr2, expr3 ) CV_Assert_2(expr1, expr2); CV_Assert_1(expr3)
-#define CV_Assert_4( expr1, expr2, expr3, expr4 ) CV_Assert_3(expr1, expr2, expr3); CV_Assert_1(expr4)
```

Comment:
We should keep order of checks like C++ `if` do.
Example:
```
CV_Assert_N(obj, obj->isValid);
```

We want exception here instead of `SIGSEGV` signal.

Reply:
Right.

No, it is just a comment for the example when it tries dereferencing NULL pointer before actual NULL check.

Reply:
@alalek Thanks for noticing that! I just fixed it.